xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -95,39 +91,21 @@ bool CRendererMediaCodecSurface::RenderCapture(CRenderCapture* capture)
   return true;
 }
 
-void CRendererMediaCodecSurface::AddVideoPicture(const VideoPicture &picture, int index)
+void CRendererMediaCodecSurface::AddVideoPicture(const VideoPicture &picture, int index, double currentClock)
 {
-  ReleaseBuffer(index);
+  int64_t nanodiff(static_cast<int64_t>((picture.pts - currentClock) * 1000));
 
-  BUFFER &buf(m_buffers[index]);
-  if (picture.videoBuffer && (buf.videoBuffer = dynamic_cast<CMediaCodecVideoBuffer*>(picture.videoBuffer)))
-    buf.videoBuffer->Acquire();
+  if (dynamic_cast<CMediaCodecVideoBuffer*>(picture.videoBuffer))
+    dynamic_cast<CMediaCodecVideoBuffer*>(picture.videoBuffer)->RenderUpdate(m_surfDestRect,
+      std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::steady_clock::now().time_since_epoch()).count() + nanodiff);
```

Comment:
Should you use CurrentHostCounter here?
It's guaranteed to to be same clock as the one used internally by Android, while I personally don't know about std::chrono::steady_clock::now().time_since_epoch()